src/gen9_vdenc.c

```diff
@@ -3120,6 +3133,9 @@ gen9_vdenc_mfx_avc_inset_headers(VADriverContextP ctx,
 
     if (vdenc_context->is_frame_level_vdenc)
         insert_one_zero_byte = 0;
+    else {
+        insert_one_zero_byte = 1;
+    }
 
```

Comment:
The flag of insert_one_zero_byte will vary on the platform.
So I would like to keep the current flowchart. When it is for is_frame_level_vdenc==0, it is configured again.

For the flag check of gen9_vdenc_mfx_avc_insert_slice_packed_data(), currently we only prepare one slice_header for every slice.  If multi data buffers are needed for one slice, it is not clear whether it is really needed.
I will double check it.